src/main/c/Objects/pyjclass.c

```diff
@@ -276,12 +325,80 @@ static PyObject* pyjclass_call(PyJClassObject *self,
     return result;
 }
 
+// get attribute 'name' for object.
+// uses obj->attr dict for storage.
+// returns new reference.
+static PyObject* pyjclass_getattro(PyObject *obj, PyObject *name)
+{
+    PyObject *ret = PyObject_GenericGetAttr(obj, name);
+    if (ret == NULL) {
+        return NULL;
+    } else if (PyJMethod_Check(ret) || PyJMultiMethod_Check(ret)) {
+        /*
+         * TODO Should not bind non-static methods to pyjclass objects, but not
+         * sure yet how to handle multimethods and static methods.
```

Comment:
So what happens at runtime with this change when you try to invoke an instance method on a pyjclass?

Reply:
Before this change:
```
>>> from java.util import ArrayList
>>> ArrayList.size
<bound method size of <jep.PyJClass object at 0x7f88ebed50b0>>
>>> ArrayList.size()
Instantiate this class before calling an object method.
>>> instance = ArrayList()
>>> instance.size
<bound method size of <ArrayList object at 0x7f88ebed83b0>>
>>> instance.size()
0
>>> pyjtype = type(instance)
>>> pyjtype
<class 'java.util.ArrayList'>
>>> pyjtype.size
type object 'ArrayList' has no attribute 'size'
>>> pyjtype.size()
type object 'ArrayList' has no attribute 'size'
```
After this change:
```
>>> from java.util import ArrayList
>>> ArrayList.size
<bound method size of <jep.PyJClass object at 0x7f6231b97730>>
>>> ArrayList.size()
Instantiate this class before calling an object method.
>>> instance = ArrayList()
>>> instance.size
<bound method size of <java.util.ArrayList object at 0x7f6231ba0f80>>
>>> instance.size()
0
>>> pyjtype = type(instance)
>>> pyjtype
<class 'java.util.ArrayList'>
>>> pyjtype.size
<jep.PyJMethod object at 0x7f6231b9f8b0>
>>> pyjtype.size()
Invalid number of arguments: 0, expected 1.
>>> pyjtype.size(instance)
0
```
This particular comment was copied along with code from pyjobject, I think it is indicating pyjclass should behave like pyjtype and return unbound methods so you could pass in an instance. The comment I added above `pyjclass_init_attr` is suggesting that instead we should just remove them entirely from the pyjclass. The two comments are contradictory but I don't think we have a definitive path forward so I don't want to get rid of the old comment.

My rational for suggesting removing things from pyjclass is that eventually I think it would be nice if pyjclass was actually exposing the reflective methods from java.lang.Class, so from Python you could do some Java style reflection with things like getDeclaredMethods() and getAnnotations(). I think it would be confusing and complicated to have all the current methods of pyjclass mixed in the same object with those.

Reply:
I like the new behavior, it behaves more like Python and is therefore better.